### DISCUSSION TITLE
Create Connections for Direct Peering Sessions?

### DISCUSSION THREAD

**Discussion by bz-ah** [peering-manager/peering-manager#831] [General]
Hi!  I am new to Peering Manager.  I am working on integrating it into our environment and am running into what I think might be a potential feature request but, because I'm new to the software, wanted to post here first.

So, you can define  Internet Exchanges and within those objects you can create Connections.  Is there any way to also do this for Direct Peering Sessions?  It seems this feature is tied into IXes.

I would really like the ability to create Connections for Direct Peering Sessions as it would be nice to be able to document the physical and/or logical interfaces associated with these sessions as well as things like circuit IDs and whatnot.  I started doing this as comments on the Direct Peering Session entry but logically this stuff belongs with the connection _associated_ with the peering session.

Not sure if I'm the only one who'd like to see the feature?  Perhaps there is a way to already do this I just haven't figured out?

**Comment by wtremmel** [comment]
> So, you can define Internet Exchanges and within those objects you can create Connections. Is there any way to also do this for Direct Peering Sessions? It seems this feature is tied into IXes.

You can also add direct peering sessions for each AS you are peering with

> 
> I would really like the ability to create Connections for Direct Peering Sessions as it would be nice to be able to document the physical and/or logical interfaces associated with these sessions as well as things like circuit IDs and whatnot.

IMHO this is (at least partly) out of scope for Peering Manager. Perhaps Netbox is better suited for this.

**Comment by bz-ah** [comment]
> > So, you can define Internet Exchanges and within those objects you can create Connections. Is there any way to also do this for Direct Peering Sessions? It seems this feature is tied into IXes.
> 
> You can also add direct peering sessions for each AS you are peering with

Yes, and that's what I've been doing.  Apologies for not making that clear.

> > I would really like the ability to create Connections for Direct Peering Sessions as it would be nice to be able to document the physical and/or logical interfaces associated with these sessions as well as things like circuit IDs and whatnot.
> 
> IMHO this is (at least partly) out of scope for Peering Manager. Perhaps Netbox is better suited for this.

I had a suspicion this may be.  My thinking was to make Peering Manager a "one stop shop" for the things related to all of our peers (IPs, ASNs, logical/phyiscal ports, circuit IDs [if applicable]).  It looks like this is 100% covered with IXes in a logical (to me) organization, but less so with direct peers.  I _can_ add information to the comment field of a given direct peering session but because that's not always 1:1 it doesn't feel like the best approach.  Like we may have two peers that share the same port (IPv4 and IPv6 for example) or we may have one peer that spans multiple ports (LAG).

We can totally document this stuff in Netbox, yeah.  With how nice we can also do this in Peering Manager with IXes I was thinking it'd be nice to be able to do the same with Direct Peers.  Of course, we could treat them like IXes but that also does not feel right, haha.

**Reply by ianbarrere** [reply]
I was actually just writing a feature request for this very thing when I thought to check the discussions as well, and found this.

I too think this is a logical extension of PeeringManager's functionality. I get that it might be outside the scope of PeeringManager, but then it seems like the connections should be scrapped altogether, for IXes as well.

I use PeeringManager as a source of truth for all our peering, and my automation relies on the connection definitions to set up the actual interfaces. In fact, our Netbox installation activates interface config by sending a webhook to our automation platform when an interface is changed, and indeed I have PeeringManager set up to adjust IX connection interface configuration in Netbox via a webhook as well, so if you make a change to a connection in PeeringManager, it changes Netbox which in turn changes the actual interface configuration on the device, but I can't do that for direct and transit peers, which to me feels like a shortcoming.

I guess it's kind of a gray area, and I could see it being entirely one way or the other, but at the moment it's half and half, which doesn't seem right.

Not sure if you ever solved this for yourself, but thought I'd mention that I'm in the same boat.
